### PR TITLE
Fix #338 Allow user-specified job name in DataflowPipelineOptions

### LINKED ISSUE #338: Set custom `JobName`
When specifying a jobName from command line the jobName seems to be is overridden by scio:

To reproduce:
```
> run-main YourJob --runner=DataflowPipelineRunner --streaming --update --jobName=JOB_NAME --etc.
[info] Running YourJob --runner=DataflowPipelineRunner --streaming ...
Dataflow SDK version: 1.8.0
[error] (run-main-1) java.lang.IllegalArgumentException: Could not find running job named topks-rcschermesser-1115160752
java.lang.IllegalArgumentException: Could not find running job named topks-rcschermesser-1115160752
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.getJobIdFromName(DataflowPipelineRunner.java:3210)
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.run(DataflowPipelineRunner.java:613)
	at com.google.cloud.dataflow.sdk.runners.DataflowPipelineRunner.run(DataflowPipelineRunner.java:200)
	at com.google.cloud.dataflow.sdk.Pipeline.run(Pipeline.java:181)
	at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:303)
	at com.spotify.scio.ScioContext$$anonfun$close$1.apply(ScioContext.scala:297)
	at com.spotify.scio.ScioContext.requireNotClosed(ScioContext.scala:343)
	at com.spotify.scio.ScioContext.close(ScioContext.scala:297)
	at algolia.harrylogger.runners.log.TopKs$.main(TopKs.scala:42)
	at algolia.harrylogger.runners.log.TopKs.main(TopKs.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
```

I suspect those lines to be the issue: https://github.com/spotify/scio/blob/master/scio-core/src/main/scala/com/spotify/scio/ScioContext.scala#L293-L294

### PR DESCRIPTION
**Pull request by andrewsmartin**
@ElPicador fyi

### CODE REVIEW

File: scio-core/src/main/scala/com/spotify/scio/ScioContext.scala [hunk 1]
```diff
@@ -87,9 +87,14 @@ object ScioContext {
     new ScioContext(options, artifacts)
 
   /** Create a new [[ScioContext]] instance for testing. */
-  def forTest(): ScioContext = {
+  def forTest(jobName: String = ""): ScioContext = {
```
Comment (andrewsmartin) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: @ravwojdyla @nevillelyh want to get your feedback especially here. jobName is a dataflow specific option so not sure if it makes sense here. Supporting user-defined job names from the test context isn't really that important and I can get rid of this if you think it's better without.

Comment (nevillelyh) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: It's not really used in any test right? In that case I prefer to not have it.

Comment (andrewsmartin) [inline scio-core/src/main/scala/com/spotify/scio/ScioContext.scala]: Yeah none until now. I'll get rid of it.

**Comment by ElPicador** [comment]
Nice, thanks 😄

**Comment by ravwojdyla** [comment]
👍

**Pull request merged**